src/contextMenu.component.ts

```diff
@@ -85,17 +92,36 @@ export class ContextMenuComponent implements AfterContentInit {
     return {
       'position': 'fixed',
       'display': this.isShown ? 'block' : 'none',
-      left: this.mouseLocation.left,
-      marginLeft: this.mouseLocation.marginLeft,
-      marginTop: this.mouseLocation.marginTop,
-      top: this.mouseLocation.top,
+      left: this.mouseLocation.left + 'px',
+      marginLeft: this.mouseLocation.marginLeft + 'px',
+      marginTop: this.mouseLocation.marginTop + 'px',
+      top: this.mouseLocation.top + 'px',
     };
   }
 
-  @HostListener('document:click')
-  @HostListener('document:contextmenu')
-  public clickedOutside(): void {
-    if (!this.isOpening) {
+  @HostListener('document:click', ['$event'])
+  @HostListener('document:contextmenu', ['$event'])
+  public clickedOutside($event: any): void {
+    let outside = true
+
+    const undefZero = (x: number) => x ? x : 0
+
+    if (this.menuElement)
+    {
+      const top = this.mouseLocation.top + undefZero( this.mouseLocation.marginTop )
+      const left = this.mouseLocation.left + undefZero( this.mouseLocation.marginLeft )
+      const outH = $event.clientX < left || // left
+                  left +
+                  this.menuElement.nativeElement.clientWidth < $event.clientX // right
+
+      const outV = $event.clientY < top || // above
+                  top +
+                  this.menuElement.nativeElement.clientHeight < $event.clientY // below
+
+      outside = outH || outV
+    }
+
+    if (!this.isOpening && outside) {
```

Comment:
Great, I will do that.

Since I don't have to do calculations anymore, should I revert my changes to MouseLocation (going from `string` to `number`)? It introduced some bugs during development (`undefined+0=undefined`...), it may be better to leave it as it was.